### ISSUE TITLE
Add ability to query some basic client info via lua

### ISSUE THREAD

**Issue by Fixer-007** [minetest/minetest#4331]
... like engine, o/s, protocol version, processor architecture

With this people can gather some statistics and even ban non-x86 players from servers, or make tablet-only server to gather all mom/dad/suicide/rape talkers.

**Comment by SmallJoker** [comment]
This would only make sense to optimize some big-screen mods like unified_inventory.
Everybody could change the source code to bypass a mod that blocks all Android/iOS clients.
Reference to (#minetest-projcet discussion)[http://minetest.org/ircproject/2016-07-18.html]

**Comment by est31** [comment]
The use case of banning people from servers based on the architecture is ridiculous. Its predjudice based and would discriminate users of an entire platform. Its also ineffective. Most of the players on non x86 come from minetest forks, and the authors may change the info to fake the x86 platform so that the players still can play.

Really, if people don't like to have their servers exposed to the public, they should not announce them on the serverlist or even add default passwords.

What's next, feature request for credit card number so that there is an 18+ check?

**Comment by est31** [comment]
TLDR :-1:

**Comment by lightonflux** [comment]
I would like to see at least these infos which can hardly be used for platform discrimination:

 * RAM
 * window size
 * MT version

These would help to answer some of the questions:

 * Should the Mt default window size be increased?
 * What space can inventory mods expect?
 * How fast do our players update?

**Comment by DonBatman** [comment]
If this was to happen I would like to see it as informing the player that their hardware or version may not meet minimum requirements for the server. Then give them the option to proceed or return to server list.

**Comment by sofar** [comment]
heh, 18+ check :D

**Comment by sofar** [comment]
Any information identifying the client properties (cpu, ram size, screen size) could be used to track players. Out of principle, I find this unacceptable in general, and plain unethical in open source software. It doesn't actually help us debug issues, either.

**Comment by sfan5** [comment]
This was already rejected by serveral core devs in the past: #1216

**Comment by rubenwardy** [comment]
Your usecase there was statistics, not device capabilities (screen size, RAM) like this thread is asking for.

Also, it's already possible to gather statistics (anonymously) using remote media

**Comment by sfan5** [comment]
What this issue requests is pretty similar and the reasons given also apply to this.

**Comment by Fixer-007** [comment]
Recent change allows to see client version - project, string, hash.
https://github.com/minetest/minetest/pull/4712

Example:
Minetest
0.4.14-569b89b
0.4.14-569b89b-45eab34-dirty

**Comment by est31** [comment]
@Fixer-007 no it shows the server version, not the client one.

Either way consensus seems to reject this change.

**Comment by Fixer-007** [comment]
oh...
